### PR TITLE
PLU-309: [TILES-ATOMIC-INCREMENT-4] backend implementation for increment/decrement row value

### PR DESCRIPTION
**Pull request by pregnantboy**
### TL;DR
Add backend implementation for 'add' and 'subtract' operations for Tiles update row action

### What changed?
- Modified `patchTableRow` to handle mathematical operations
  - added tests
  - error handling for invalid numeric operations
- Updated column metadata for tiles action to maintain column ordering
  - Fixed variable sorting logic for null order values

### How to test?
1. Create a tile with a few columns and values that are both numbers and strings
2. Test updating cells using:
   - normal updates (set as)
   - adding
   - subtracting
   - mixed
3. Verify error handling by:
   - Attempting math operations on non-numeric values (both original values and operands)
4. Check that columns maintain their order in the UI

### Regression test
1. Check that existing tiles update rows are still working

### CODE REVIEW

**Comment by pregnantboy** [comment]
* **#835** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/835?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/835?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#819** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/819?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#818** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/818?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#817** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/817?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/frontend/src/helpers/variables.ts [hunk 1]
```diff
@@ -32,13 +32,13 @@ function sortVariables(variables: Variable[]): void {
   variables.sort((a, b) => {
     // Put vars with null order last, but preserve ordering (via `sort`'s
     // stability) if both are null.
-    if (!a.order && !b.order) {
+    if (a.order == null && !b.order == null) {
```
Comment (m0nggh) [inline packages/frontend/src/helpers/variables.ts]: `b.order == null` instead

**Review by m0nggh — APPROVED** [review]
other than the small nit, lgtm

tested that the
- 3 operations work
- error is caught for non-numeric add/subtract
- old update row steps still work
- columns variables now maintain order

File: packages/frontend/src/helpers/variables.ts (hunk 1 above)
Comment (pregnantboy) [reply packages/frontend/src/helpers/variables.ts]: oops!

Comment (pregnantboy) [reply packages/frontend/src/helpers/variables.ts]: goood catch

**Comment by pregnantboy** [comment]
### Merge activity

* **Mar 10, 2:47 AM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/835).
* **Mar 10, 2:49 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/835) rebased this pull request as part of a merge.
* **Mar 10, 2:51 AM EDT**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/835).
